services/wallet/activity/session_service.go

```diff
@@ -65,13 +64,62 @@ func (s *Service) getActivityEntries(ctx context.Context, f fullFilterParams, of
 	if f.version != V2 {
 		return nil, errors.New("unsupported version")
 	}
-	allAddresses := s.areAllAddresses(f.addresses)
 
-	entries, err := getActivityEntriesV2(ctx, s.getDeps(), f.addresses, allAddresses, f.chainIDs, f.filter, offset, count)
+	// Get fetched activities from Alchemy
+	fetchedEntries, err := getFetchedActivitiesEntries(ctx, s.getDeps(), f.addresses, f.chainIDs, offset, count)
 	if err != nil {
-		fmt.Println("error getting activity entries", err)
+		return nil, err
 	}
-	return entries, nil
+
+	// Extract transaction IDs from fetched entries
+	txIDs := make([]TransactionID, 0)
+	for _, entry := range fetchedEntries {
+		for _, tx := range entry.transactions {
+			txID := TransactionID{
+				ChainID: tx.ChainID,
+				Hash:    tx.Hash,
+			}
+			txIDs = append(txIDs, txID)
+		}
+	}
+
+	// Get saved transactions by hash
+	savedEntriesMap, err := getSentActivitiesByHash(ctx, s.getDeps(), f.addresses, f.chainIDs, txIDs)
+	if err != nil {
+		// Continue with fetched entries even if we can't get saved ones
+		savedEntriesMap = make(map[TransactionID]Entry)
+	}
+
+	// Build result array: use saved entry if exists, otherwise use fetched entry
+	resultEntries := make([]Entry, 0, len(fetchedEntries))
+
+	for _, fetchedEntry := range fetchedEntries {
```

Comment:
If I'm not mistaken, this logic implies sent Txs won't show up in the results until those transactions are fetched from Alchemy. Is that correct? I don't think that's desired behavior.

Reply:
Example: we want sent entries to show up as soon as we perform the tx. Alchemy will probably not give them to us until the txs are confirmed.